src/ORM/Behavior/EnumBehavior.php

```diff
@@ -0,0 +1,125 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Behavior;
+
+use ArrayObject;
+use BackedEnum;
+use Cake\Datasource\EntityInterface;
+use Cake\Event\EventInterface;
+use Cake\ORM\Behavior;
+use Cake\ORM\Entity;
+use Cake\ORM\Query;
+
+/**
+ * Enum behavior
+ *
+ * Allows you to map a specific field in your model to a backed enum
+ *
+ *  * Add behavior + config to connect a field with a backed enum
+ * ```
+ * $this->addBehavior('Enum', [
+ *     'fieldMap' => [
+ *         'published' => ArticleStatus::class
+ *     ]
+ * ]);
+ * ```
+ */
+class EnumBehavior extends Behavior
+{
+    /**
+     * Default config
+     *
+     * These are merged with user-provided configuration when the behavior is used.
+     *
+     * @var array<string, mixed>
+     */
+    protected array $_defaultConfig = [
+        'fieldMap' => [],
+    ];
+
+    /**
+     * Transform array data into entity data
+     *
+     * @param \Cake\Event\EventInterface $event The beforeMarshal event that was fired
+     * @param \ArrayObject $data Data present before converting to an entity
+     * @param \ArrayObject $options Options passed to the event
+     * @return void
+     */
+    public function beforeMarshal(EventInterface $event, ArrayObject $data, ArrayObject $options): void
+    {
+        $fieldMap = $this->getConfig('fieldMap');
+        /** @var \BackedEnum $enumType */
+        foreach ($fieldMap as $field => $enumType) {
+            if ($data->offsetExists($field)) {
+                $value = $data->offsetGet($field);
+                if (is_int($value) || is_string($value)) {
+                    $data->offsetSet($field, $enumType::from($value));
```

Comment:
Should the behavior/Validation also provide methods for validating enum values? Providing methods/wrappers for validation would help ensure that we don't hit exceptions when values that are not actually in the enum.

Reply:
Curently I only set the associated field to invalid with a generic error message.
But I guess you mean like providing a more user-facing interface (or config) to define how the validation is being done.